### PR TITLE
Updated award queue to use raw zap.Logger rather than Sugar'd version

### PR DESCRIPTION
**Pull request by ntwyman**
## Description

Backed out the use of the Sugar'd Logger in favor of the raw, structured logger which should give more machine friendly logs.  

Along the way I also added func MarshalLogObject(encoder)... to the TPS and TPL structures. This lets them be logged natively using zap.Object.

## Reviewer Notes

Check that I didn't accidentally break anything else.

## Verification Steps

* [*] All tests pass.
* [*] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [n/a] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [n/a] There are no [aXe](https://www.deque.com/products/aXe/) warnings for UI.
* [n/a] This works in IE.
* [n/a] Any new third party client side dependencies (e.g., login.gov, google analytics, CDN libraries, etc.) have been communicated to @willowbl00.

### CODE REVIEW

File: pkg/awardqueue/awardqueue.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ func (aq *AwardQueue) attemptShipmentOffer(shipment models.ShipmentWithOffer) (*
 		tspPerformance, err := models.NextEligibleTSPPerformance(aq.db, tdl.ID, shipment.BookDate)
 
 		if loopCount != 0 && tspPerformance.ID == firstTSPid {
-			return nil, fmt.Errorf("Could not find a TSP without blackout dates in %d tries", loopCount)
+			return nil, fmt.Errorf("could not find a TSP without blackout dates in %d tries", loopCount)
```
Comment (kilbergr) [inline pkg/awardqueue/awardqueue.go]: Why this change? Also, do you not want to change this into an aq.logger.Error instance?

File: pkg/awardqueue/awardqueue.go [hunk 2]
```diff
@@ -123,14 +123,14 @@ func (aq *AwardQueue) assignShipments() {
...
 			_, err = aq.attemptShipmentOffer(shipment)
 			if err != nil {
-				aq.logger.Errorf("Failed to offer shipment: %s", err)
+				aq.logger.Error("Failed to offer shipment", zap.Error(err))
 			} else {
 				count++
 			}
 		}
-		aq.logger.Infof("Awarded %d shipments.", count)
+		aq.logger.Info("Awarded some shipments.", zap.Int("count", count))
```
Comment (kilbergr) [inline pkg/awardqueue/awardqueue.go]: Maybe make this "total_count" or something to indicate it's not just a single TSP count?

File: pkg/awardqueue/awardqueue.go [hunk 3]
```diff
@@ -76,7 +76,7 @@ func (aq *AwardQueue) attemptShipmentOffer(shipment models.ShipmentWithOffer) (*
 		err = aq.db.Transaction(func(tx *pop.Connection) error {
 			tsp := models.TransportationServiceProvider{}
 			if err := aq.db.Find(&tsp, tspPerformance.TransportationServiceProviderID); err == nil {
-				aq.logger.Infof("Attempting to offer to TSP: %s", tsp.Name)
+				aq.logger.Info("Attempting to offer to TSP", zap.Object("tsp", tsp))
```
Comment (kilbergr) [inline pkg/awardqueue/awardqueue.go]: Make this "tsp" capital if "TDL" is capital, or change TDL to lowercase.
Why do you record shipment_id elsewhere and entire TSP here (it shows up as `"tsp":{"id":"44773adb-8219-4d08-baa1-7bf06a420dd9","name":"Excellent TSP with Blackout Date"}` anyway)? I would consider making them both ids unless there's a rationale not to.

File: pkg/awardqueue/awardqueue.go [hunk 4]
```diff
@@ -189,7 +189,7 @@ func (aq *AwardQueue) assignPerformanceBandsForTDL(tdl models.TrafficDistributio
 	for band, count := range bands {
 		for i := 0; i < count; i++ {
 			performance := perfs[perfsIndex]
-			aq.logger.Infof("Assigning tspp %s to band %d", performance.ID, band+1)
+			aq.logger.Info("Assigning tsp to band", zap.Any("tdp_id", performance.ID), zap.Int("band", band+1))
```
Comment (kilbergr) [inline pkg/awardqueue/awardqueue.go]: This was intentionally tspp (was written before we made naming standards)--stands for tspPerformance. Would you change to tspPerformance instead please?

File: pkg/awardqueue/awardqueue.go (hunk 2 above)
Comment (ntwyman) [reply pkg/awardqueue/awardqueue.go]: done

File: pkg/awardqueue/awardqueue.go (hunk 4 above)
Comment (ntwyman) [reply pkg/awardqueue/awardqueue.go]: done

File: pkg/awardqueue/awardqueue.go (hunk 1 above)
Comment (kilbergr) [reply pkg/awardqueue/awardqueue.go]: Thanks!

File: pkg/awardqueue/awardqueue.go (hunk 4 above)
Comment (kilbergr) [reply pkg/awardqueue/awardqueue.go]: introduced a typo--should be tsp_id. Can you change to tspPerformance?
